cmd/kubeadm/app/cmd/phases/kubelet.go

```diff
@@ -0,0 +1,176 @@
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+	cmdutil "k8s.io/kubernetes/cmd/kubeadm/app/cmd/util"
+	kubeletphase "k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	nodeutil "k8s.io/kubernetes/pkg/util/node"
+	"k8s.io/kubernetes/pkg/util/normalizer"
+)
+
+var (
+	kubeletWriteInitConfigLongDesc = normalizer.LongDesc(`
+		Write init kubelet configuration to disk for dynamic kubelet configuration feature, functionally equivalent to what implemented by kubeadm init.
+		` + cmdutil.AlphaDisclaimer)
+
+	kubeletWriteInitConfigExample = normalizer.Examples(`
+		# Write init kubelet configuration to disk.
+		kubeadm alpha phase kubelet write-init-config
+		`)
+
+	kubeletUploadDynamicConfigLongDesc = normalizer.LongDesc(`
+		Upload dynamic kubelet configuration to apiserver as ConfigMap, functionally equivalent to what implemented by kubeadm init.
+		` + cmdutil.AlphaDisclaimer)
+
+	kubeletUploadDynamicConfigExample = normalizer.Examples(`
+		# Upload dynamic kubelet configuration as ConfigMap.
+		kubeadm alpha phase kubelet upload-dynamic-config
+		`)
+
+	kubeletEnableDynamicConfigLongDesc = normalizer.LongDesc(`
+		Enable dynamic kubelet configuration on node, functionally equivalent to what implemented by kubeadm join. This should be run on nodes.
+		` + cmdutil.AlphaDisclaimer)
+
+	kubeletEnableDynamicConfigExample = normalizer.Examples(`
+		# Enable dynamic kubelet configuration on node.
+		kubeadm alpha phase kubelet enable-dynamic-config
+		`)
+)
+
+// NewCmdKubelet returns main command for Kubelet phase
+func NewCmdKubelet() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "kubelet",
+		Short: "Adopts dynamic kubelet configuration.",
+		Long:  cmdutil.MacroCommandLongDescription,
+	}
+
+	cmd.AddCommand(NewCmdKubeletWriteInitConfig())
+	cmd.AddCommand(NewCmdKubeletUploadDynamicConfig())
+	cmd.AddCommand(NewCmdKubeletEnableDynamicConfig())
+
+	return cmd
+}
+
+// NewCmdKubeletWriteInitConfig calls cobra.Command for writing init kubelet configuration
+func NewCmdKubeletWriteInitConfig() *cobra.Command {
+	var cfgPath string
+	cmd := &cobra.Command{
+		Use:     "write-init-config",
+		Short:   "Write init kubelet configuration to disk",
+		Long:    kubeletWriteInitConfigLongDesc,
+		Example: kubeletWriteInitConfigExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := &kubeadmapiext.MasterConfiguration{}
+			legacyscheme.Scheme.Default(cfg)
+
+			// This call returns the ready-to-use configuration based on the configuration file that might or might not exist and the default cfg populated by flags
+			internalcfg, err := configutil.ConfigFileAndDefaultsToInternalConfig(cfgPath, cfg)
```

Comment:
If I'm not mistaken, KubernetesVersion is not used by kubelet phase, so it is recommended to set this explicitly to some costant value in order to avoid the lookup of the version from the internet when executing `ConfigFileAndDefaultsToInternalConfig`. See e.g. [here](https://github.com/kubernetes/kubernetes/blob/754bb1350faa34736de9216310dd00940bc56d8a/cmd/kubeadm/app/cmd/phases/bootstraptoken.go#L113)
The same applies below

Reply:
Agreed. Will do.

Reply:
done.